synapse/server.py

```diff
@@ -216,6 +216,13 @@ def get_reactor(self):
         """
         return self._reactor
 
+    def get_sendmail(self):
+        if hasattr(self, "_sendmail"):
+            return self._sendmail
+        else:
+            from twisted.mail.smtp import sendmail
+            return sendmail
```

Comment:
Can we move the import up to the top please?

Also, I'd be tempted to instead add `sendmail` to `DEPENDENCIES` and define:

```python
def build_sendmail(self):
    return sendmail
```

Which bring it inline with other dependencies. To override in the tests its enough to explicitly define `sendmail` on the HS object, which can be done by passing `sendmail` kwarg to `setup_test_homeserver`